### PR TITLE
implement TimedeltaArray asm8, to_timedelta64

### LINKED ISSUE #22632: TimedeltaIndex should have to_timedelta64 mirroring Timedelta
Also `asm8`

### PR DESCRIPTION
**Pull request by jbrockmendel**
For compat with Timedelta scalar

- [x] closes #22632
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/timedeltas.py [hunk 1]
```diff
@@ -355,6 +355,31 @@ def to_pytimedelta(self):
...
+
+        Notes
+        -----
+        This returns a view on self, not a copy.
+
+        See also
+        --------
+        Timedelta.to_timedelta64
+        """
+        return self.asi8.view('m8[ns]')
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/timedeltas.py]: you are going a little bit in circles here, as `asi8` itself is already `self.values.view('i8')` ?

File: pandas/core/arrays/timedeltas.py [hunk 2]
```diff
@@ -355,6 +355,31 @@ def to_pytimedelta(self):
...
+
+    @property
+    def asm8(self):
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/timedeltas.py]: do we need both `to_timedelta64` and `asm8` ?

**Review by jorisvandenbossche — COMMENTED** [review]
Just a general comment: for Timestamp / DatetimeIndex you have the same inconsistency of `to_datetime64`. 
Also on the Index/Array classes, you already have other ways to get the numpy array (`np.array(..)`, although might sometimes give object dtype). Mainly wondering if we should have a more general API across the arrays to convert to certain numpy types.

File: pandas/core/arrays/timedeltas.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/timedeltas.py]: I find `asi8` much more explicit than alues (BTW as I mentioned to jreback the "v" key on my keyboard is sticky; I'm going to stop copy-pasting and let you guys infer the missing letter for a while).

File: pandas/core/arrays/timedeltas.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/timedeltas.py]: If we were starting fresh I would say no, but as it is I think consistency with the scalar type is really worthwhile.  (and to your point aboe DatetimeIndex should have a `to_datetime64` method)

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
Updated to implement DatetimeArray.asm8, DatetimeArray.to_datetime64

**Comment by jorisvandenbossche** [comment]
Thinking about this a bit more: I am -1 on adding duplicate attributes/methods just for sake of consistency with our scalar type `Timestamp`/`Timedelta`. We already have a bloated API.

Also for DatetimeIndex/TimedeltaIndex, it is not really clear to me what this is targeted use case is, as it just returns a copy of itself? (except that it returns an array instead of index? Which I also don't find clear from the name)

**Comment by jbrockmendel** [comment]
> Thinking about this a bit more: I am -1 on adding duplicate attributes/methods just for sake of consistency with our scalar type Timestamp/Timedelta. We already have a bloated API.

Would you advocate removing one/both of the attributes from the Timedelta/Timestamp scalars?

**Comment by jreback** [comment]
thanks!

**Comment by jorisvandenbossche** [comment]
Sorry, Jeff, but I clearly objected this PR. Can you then at least discuss it before merging? 
So I reverted the PR.

**Comment by jreback** [comment]
@jorisvandenbossche if u object then your need to actually put in a review request to do so

**Pull request merged**